docs/DeveloperGuide.md

```diff
@@ -215,18 +215,143 @@ storage.saveAllData();
 
 <ins>Overview</ins>
 
-<ins>Implementation Details</ins>
+The `Group` class is used to represent a group of people who have transactions among themselves. The `GroupList` class is used to represent a list of groups
+stored in the application.
 
 <ins>Class Structure</ins>
 
+The `Group` class has the following attributes:
+* *memberList*: A MemberList object representing the list of Members in the group.
+* *transactionList*: A TransactionList object representing the list of Transactions in the group.
+* *storage*: A StorageHandler object representing the storage handler for the group.
+* *groupName*: A string representing the name of the group.
+* *transactionSolution*: An array list collection of Subtransaction objects representing the least transactions solution to solving all debts in the group.
+
+The `GroupList` class has the following attributes:
+* *activeGroup*: A Group object representing the currently active group.
+* *groupList*: An array list collection of Group objects representing the list of groups stored in the application.
+
+
+<ins>Implementation Details</ins>
+
+The detailed class diagram for `Group` and `GroupList` can be found below.
+
+![Group Class Diagram](diagrams/Group Class.png)
```

Comment:
Image is broken
![image](https://github.com/AY2324S2-CS2113-T15-1/tp/assets/13176823/e4ab706f-7b42-4910-9667-573cfe4c65dc)